.github/workflows/rust.yml

```diff
@@ -0,0 +1,55 @@
+name: Rust
+
+on:
+  push:
+    branches: [ main ]
+  schedule:
+    - cron: 0 0 1 * *
+  pull_request:
+    branches: [ main ]
+
+env:
+  CARGO_INCREMENTAL: 1
```

Comment:
Pretty sure it's already on by default for debug builds https://doc.rust-lang.org/cargo/reference/profiles.html#default-profiles

Reply:
rust-toolchain disable it so we need to add it explicitly
```console
Run : disable incremental compilation
  : disable incremental compilation
  if [ -z "${CARGO_INCREMENTAL+set}" ]; then
    echo CARGO_INCREMENTAL=0 >> $GITHUB_ENV
  fi
  shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
  env:
    CARGO_INCREMENTAL: 0
    CACHE_ON_FAILURE: true
    ```

Reply:
Maybe it would be easier to just do:
```
run: rustup update
```
it's a one liner and it wont disable cargo incremental

Reply:
Maybe do you have an idea on how I can remove these warnings?
```console
warning: tool `rust-analyzer` is already installed, remove it from `/home/runner/.cargo/bin`, then run `rustup update` to have rustup manage this tool.
warning: tool `rustfmt` is already installed, remove it from `/home/runner/.cargo/bin`, then run `rustup update` to have rustup manage this tool.
warning: tool `cargo-fmt` is already installed, remove it from `/home/runner/.cargo/bin`, then run `rustup update` to have rustup manage this tool.
```
But seems to work fine now

Reply:
Not really. tbh I'm not sure doing rustup update is even very useful. By default the worker is on stable and it is relatively up to date. rustup update make sure it is really up to date but that's not absolutely necessary. If you run 1.61 instead of 1.64, who cares anyway

Reply:
Good enough then :rocket: